### PR TITLE
Add module testing steps

### PR DESCRIPTION
**Pull request by kaiguo**

### CODE REVIEW

File: docs/native-modules-setup.md [hunk 1]
```diff
@@ -177,4 +177,24 @@ To make sure that everything is working, you'll want to try building `MyLibrary`
...
 
-You have now created the scaffolding to build a native module or view manager. Now it's time to add the business logic to the module - follow the steps described in the [Native Modules](native-modules.md) and [View Managers](view-managers.md) documents.
\ No newline at end of file
+You have now created the scaffolding to build a native module or view manager. Now it's time to add the business logic to the module - follow the steps described in the [Native Modules](native-modules.md) and [View Managers](view-managers.md) documents.
+
+### Testing the module before it gets published
+
+#### Option 1: Create a new test app
+1. Follow the [getting started guide](getting-started.md) to create a new React Native Windows app.
+2. Run `npm i <module-local-path> --save` to install the local module.
```
Comment (stmoy) [inline docs/native-modules-setup.md:186]: Can you include an example of what this path looks like, assuming the steps above referencing `MyLibrary`?

File: docs/native-modules-setup.md [hunk 2]
```diff
@@ -177,4 +177,24 @@ To make sure that everything is working, you'll want to try building `MyLibrary`
...
+3. [Link the native module](native-modules-using.md).
+
+#### Option2: Adding Windows support to existing sample app.
```
Comment (stmoy) [inline docs/native-modules-setup.md:189]:
```suggestion
#### Option 2: Adding Windows support to existing sample app

```

File: docs/native-modules-setup.md [hunk 3]
```diff
@@ -177,4 +177,24 @@ To make sure that everything is working, you'll want to try building `MyLibrary`
...
+If you are working on an existing module that already has iOS and Android samples, and want to add window part in the existing test app, follow these steps.
```
Comment (stmoy) [inline docs/native-modules-setup.md:190]:
```suggestion
If you are working on an existing module that already has iOS and Android samples, and want to add Windows support to the existing test app, follow these steps.

```

File: docs/native-modules-setup.md [hunk 4]
```diff
@@ -177,4 +177,24 @@ To make sure that everything is working, you'll want to try building `MyLibrary`
...
+1. In a different directory, follow the [getting started guide](getting-started.md) and create a new React Native Windows app. Copy `windows` folder into the existing sample app folder.
```
Comment (stmoy) [inline docs/native-modules-setup.md:191]:
```suggestion
1. In a different directory, follow the [getting started guide](getting-started.md) and create a new React Native Windows app. 

1. Copy the `Windows` folder from the module into the sample app's folder.

```

File: docs/native-modules-setup.md [hunk 5]
```diff
@@ -177,4 +177,24 @@ To make sure that everything is working, you'll want to try building `MyLibrary`
...
+2. Open `sln` and `vxcproj` files and check `node_module` reference paths. Fix the paths if necessary based on how the folders are structured in native module repo ([example](https://github.com/react-native-community/react-native-webview/blob/master/example/windows/WebViewWindows.sln#L11-L42)).
+3. Open the solution with VisualStudio and [link native module](native-modules-using.md).
```
Comment (stmoy) [inline docs/native-modules-setup.md:193]:
```suggestion
3. Open the solution with Visual Studio and [link native module](native-modules-using.md).

```

File: docs/native-modules-setup.md (hunk 4 above)
Comment (stmoy) [reply docs/native-modules-setup.md:191]: Why do they need to follow the "getting started guide" to create a new React Native Windows app if they are trying to add support to an existing app?

File: docs/native-modules-setup.md [hunk 6]
```diff
@@ -177,4 +177,24 @@ To make sure that everything is working, you'll want to try building `MyLibrary`
...
+
+#### Option 1: Create a new test app
+1. Follow the [getting started guide](getting-started.md) to create a new React Native Windows app.
+2. Run `npm i <module-local-path> --save` to install the local module.
+3. [Link the native module](native-modules-using.md).
+
+#### Option2: Adding Windows support to existing sample app.
+If you are working on an existing module that already has iOS and Android samples, and want to add window part in the existing test app, follow these steps.
+1. In a different directory, follow the [getting started guide](getting-started.md) and create a new React Native Windows app. Copy `windows` folder into the existing sample app folder.
+2. Open `sln` and `vxcproj` files and check `node_module` reference paths. Fix the paths if necessary based on how the folders are structured in native module repo ([example](https://github.com/react-native-community/react-native-webview/blob/master/example/windows/WebViewWindows.sln#L11-L42)).
```
Comment (stmoy) [inline docs/native-modules-setup.md:192]: Please copy the sample into the doc to make it easier to follow.

File: docs/native-modules-setup.md (hunk 4 above)
Comment (kaiguo) [reply docs/native-modules-setup.md:193]: They need the content in `windows` folder from the new app.

Ideally they should be able to just run `npx react-native-windows-init` to generate `windows` folder, but I think the command is looking for `package.json`, which is not located inside the example folder in most native module setups, so we have to create a new project somewhere else and then copy `windows` over manually :\

Comment (stmoy) [reply docs/native-modules-setup.md:193]: Gotcha. My above suggestion is incorrect then, right?


```suggestion
1. In a different directory, follow the [getting started guide](getting-started.md) and create a new React Native Windows app. 

1. Copy the `Windows` folder from the blank RNW app into the existing sample app's sample app's folder. (The RNW CLI helps create the correct project setup that you can then copy directly into the sample app.)
```

**Pull request merged**